File: project_furniture/lib/models/product.dart
class Product {
  final int id;
  final String title;
  final String desc;
  final double price;
  final String image;

  Product({
    required this.id,
    required this.title,
    required this.desc,
    required this.price,
    required this.image,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      title: json['title'],
      desc: json['desc'],
      price: json['price'].toDouble(),
      image: json['image'],
    );
  }
}

File: api_laravel/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use App\Models\Category;

class Product extends Model
{
    use HasFactory;
    protected $fillable = ['title','desc','price','image'];


    public function categories(): BelongsTo
    {
        return $this->belongsTo(Category::class, 'category_id', 'id');
    }
}


